frontend/src/modules/fluenteditor/components/RichEditor.js

```diff
@@ -82,17 +82,13 @@ export default class RichEditor extends React.Component<Props> {
     ) => {
         const props = this.props;
 
-        if (!translation) {
```

Comment:
First of all, `updateUnsavedChanges()` is always called without arguments, so we should:
1. Remove arguments from the function signature.
2. Set the default values.
3. Only serialize content when we know it's serializable (and explain that in a comment).

A possible solution:
```js
updateUnsavedChanges = () => {
    const props = this.props;

    let translation = props.editor.translation;
    let initial = props.editor.initialTranslation;

    // If we're switching from or to a simple editor, we will not be able
    // to serialize the current or initial translation
    try {
        translation = fluent.serializer.serializeEntry(translation);
    } catch (e) {}

    try {
        initial = fluent.serializer.serializeEntry(initial);
    } catch (e) {}

    props.updateUnsavedChanges(translation, initial);
};
```